### PR TITLE
ASAP:Int2019 80 fix view profile page

### PR DESCRIPTION
**Pull request by robertpin**

### CODE REVIEW

File: ClientApp/viking-vault/src/UserDashboard/ProfilePage.css [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-    position: absolute;
-    background-color:white;
-    margin: 15px auto;
-    width: 65%;
-    height: 60%;
-    border-radius : 25px;
-    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
+.profile-page {
+    margin: auto;
+    width: 100%;
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/ProfilePage.css:3]: You have both width and padding, width alone is 100%, with the padding included the width will be 104%. Is this wanted behavior or and/or you have box-sizing set to border-box?

File: ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx [hunk 2]
```diff
@@ -261,24 +261,22 @@ class AccountPage extends React.Component<any, IAccountPageState>{
...
-                <div className="account-view"> 
-                    { this.state.redirect? <Redirect to = "/login"  /> : null}       
-                    {this.state.accountInfo.isPresent ?  this.showUnavailableCard() : this.showCardInformations()}
-                    <div className="accounts-information">
-                        <div className="accounts-title">
-                            <h2 className="accounts-header">Accounts</h2>
-                        </div>
-                        {this.state.accountInfo.isPresent ? this.inexistentCardNotification() : this.accountsInformation()}
+            <div className="account-view"> 
+                { this.state.redirect? <Redirect to = "/login"  /> : null}
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx]: You have two `{this.state.redirect? <Redirect to="/login"/> : null}` in your `render()`, please remove one

File: ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx [hunk 3]
```diff
@@ -261,24 +261,22 @@ class AccountPage extends React.Component<any, IAccountPageState>{
...
-                    <ResponseModal open = {this.state.openBlockCardResponseModal}  closeModal = {this.closeBlockCardModal} message = {this.state.blockCardResponseMessage}/>
+                    {this.state.accountInfo.isPresent ? this.inexistentCardNotification() : this.accountsInformation()}
+                </div>
+                {this.state.redirect? <Redirect to="/login"/> : null}
+                <label className={this.state.accountInfo.isCardBlocked ? "blocked-card-label" : "not-displayed-label"}>Your card is blocked!</label>
+                <div className = "block-card-toggle-position"> 
+                    {!this.state.accountInfo.isPresent ? <ToggleBlockCard toggleSwitch = {this.blockCard} isCardBlocked = {this.state.accountInfo.isCardBlocked}/> : null} 
                 </div>
+                <label className={this.state.accountInfo.isPresent ? "not-displayed-label" : (this.state.accountInfo.isCardBlocked ? "not-displayed-label" : "unblocked-card-label")}>Your card is unblocked!</label>
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx]: Nested ternary operators are hard to read, can you move this logic outside the return JSX (you can keep it inside the render alternatively, but before the return)

File: ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx [hunk 4]
```diff
@@ -261,24 +261,22 @@ class AccountPage extends React.Component<any, IAccountPageState>{
 
     render(){
         return(
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx:283]: Way too much logic is done inside the JSX itself, try moving some of the ternary operators outside of it. (as mentioned below for the nested ternary operators)

File: ClientApp/viking-vault/src/UserDashboard/ProfilePage.css (hunk 1 above)
Comment (robertpin) [reply ClientApp/viking-vault/src/UserDashboard/ProfilePage.css:3]: Intended behavior. It is a div in other div and it needs the padding to look good

File: ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx (hunk 4 above)
Comment (robertpin) [reply ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx:283]: Done

**Pull request merged**